File: BIL395_HW2/calculator.pl
#!/usr/bin/perl
use strict;
use warnings;

print "Simple Calculator\n";
print "----------------------\n";

while (1) {
    print "\nChoose operation:\n";
    print "1. Addition\n";
    print "2. Subtraction\n";
    print "3. Multiplication\n";
    print "4. Division\n";
    print "5. Exit\n";
    print "Enter choice (1-5): ";
    
    my $choice = <STDIN>;
    chomp $choice;
    
    last if $choice == 5;
    
    if ($choice < 1 || $choice > 4) {
        print "Invalid choice. Please try again.\n";
        next;
    }
    
    print "Enter first number: ";
    my $num1 = <STDIN>;
    chomp $num1;
    
    print "Enter second number: ";
    my $num2 = <STDIN>;
    chomp $num2;
    
    if ($choice == 1) {
        print "Result: ", $num1 + $num2, "\n";
    }
    elsif ($choice == 2) {
        print "Result: ", $num1 - $num2, "\n";
    }
    elsif ($choice == 3) {
        print "Result: ", $num1 * $num2, "\n";
    }
    elsif ($choice == 4) {
        if ($num2 == 0) {
            print "Error: Division by zero!\n";
        } else {
            print "Result: ", $num1 / $num2, "\n";
        }
    }
}

File: BIL395_HW2/calculator.rs
use std::io;

fn main() {
    println!("Simple Calculator");
    println!("Available operations: +, -, *, /");
    println!("Enter 'quit' to exit");

    loop {
        println!("Please enter your calculation (e.g., 2 + 2):");

        let mut input = String::new();
        io::stdin()
            .read_line(&mut input)
            .expect("Failed to read input");

        if input.trim().eq_ignore_ascii_case("quit") {
            println!("Goodbye!");
            break;
        }


        let parts: Vec<&str> = input.trim().split_whitespace().collect();
        if parts.len() != 3 {
            println!("Invalid input format. Please use: NUMBER OPERATOR NUMBER");
            continue;
        }


        let num1: f64 = match parts[0].parse() {
            Ok(n) => n,
            Err(_) => {
                println!("Invalid first number");
                continue;
            }
        };

        let num2: f64 = match parts[2].parse() {
            Ok(n) => n,
            Err(_) => {
                println!("Invalid second number");
                continue;
            }
        };

        let result = match parts[1] {
            "+" => num1 + num2,
            "-" => num1 - num2,
            "*" => num1 * num2,
            "/" => {
                if num2 == 0.0 {
                    println!("Error: Division by zero");
                    continue;
                }
                num1 / num2
            }
            _ => {
                println!("Invalid operator. Use +, -, *, or /");
                continue;
            }
        };

        println!("Result: {} {} {} = {}\n", num1, parts[1], num2, result);
    }
}